Holodeck/Command.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+from .Agents import *
+
+
+class Commands(object):
+    def __init__(self):
+        self._commands = []
+
+    def add_command(self, command):
+        self._commands.append(command)
+
+    def to_json(self):
+        to_return = "{\"commands\": ["
+        for x in self._commands:
+            to_return += x.to_json() + ","
+        if len(self._commands) > 0:
+            to_return = to_return[:-1]
+        to_return += "]}"
+        return to_return
+
+
+class Command(object):
+
+    def __init__(self):
+        self._number_parameters = []
+        self._string_parameters = []
+        self._command_type = ""
+
+    def set_command_type(self, command_type):
+        self._command_type = command_type
+
+    def add_number_parameters(self, number):
+        if type(number) == list:
+            self._number_parameters.extend(number)
+            return
+        self._number_parameters.append(number)
+
+    def add_string_parameters(self, string):
+        if type(string) == list:
+            self._string_parameters.extend(string)
+            return
+        self._string_parameters.append(string)
+
+    def to_json(self):
+        to_return = "{ "
```

Comment:
Joins are more efficient for string concatenation, and will help reduce for loops. Could you rewrite using joins please.